scripts/build-image.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+set -e
+
+retry() {
+    local command="$1"
+    local wait_message="$2"
+    local max_try="$3"
+
+    local attempt=0
+
+    while ! $command
+    do
+        attempt=$((attempt + 1))
+        if [[ attempt -ge $max_try ]]; then
+            return
+        fi
+
+        echo "$wait_message"
+        sleep 10
+    done
+}
+
+cleanup() {
+    local test_command="$1"
+    local clean_up_command="$2"
+    local wait_message="$3"
+    local max_try="$4"
+
+    local attempt=0
+
+    while bash -c "$test_command"
+    do
+        echo "$wait_message"
+
+        $clean_up_command
+
+        attempt=$((attempt + 1))
+        if [[ attempt -ge $max_try ]]; then
+            # Cleanup failure.
+            return 1
+        fi
+
+        sleep 10
+    done
+}
+
+cleanup '/snap/bin/lxc info builder &> /dev/null' '/snap/bin/lxc delete builder --force' 'Cleanup LXD VM of previous run' 10
+
+if [[ "$1" == "test" ]]; then
+    retry '/snap/bin/lxc launch ubuntu-daily:jammy builder --device root,size=5GiB' 'Starting LXD VM'
+else
+    retry '/snap/bin/lxc launch ubuntu-daily:jammy builder --vm --device root,size=5GiB' 'Starting LXD VM'
+fi
+retry '/snap/bin/lxc exec builder -- /usr/bin/who' 'Wait for lxd agent to be ready' 30
+retry '/snap/bin/lxc exec builder -- /usr/bin/nslookup github.com' 'Wait for network to be ready' 30
+
+/snap/bin/lxc exec builder -- /usr/bin/apt-get update
+/snap/bin/lxc exec builder --env DEBIAN_FRONTEND=noninteractive -- /usr/bin/apt-get upgrade -yq
+/snap/bin/lxc exec builder --env DEBIAN_FRONTEND=noninteractive -- /usr/bin/apt-get install linux-generic-hwe-22.04 -yq
+
+/snap/bin/lxc restart builder
+retry '/snap/bin/lxc exec builder -- /usr/bin/who' 'Wait for lxd agent to be ready' 30
+retry '/snap/bin/lxc exec builder -- /usr/bin/nslookup github.com' 'Wait for network to be ready' 30
+
+/snap/bin/lxc exec builder -- /usr/bin/apt-get update
+/snap/bin/lxc exec builder --env DEBIAN_FRONTEND=noninteractive -- /usr/bin/apt-get upgrade -yq
+/snap/bin/lxc exec builder --env DEBIAN_FRONTEND=noninteractive -- /usr/bin/apt-get install docker.io npm python3-pip shellcheck jq wget yarn -yq
```

Comment:
The `yarn` version installed by apt is not the `yarn` we expect, it is another program used to test scenarios (https://manpages.ubuntu.com/manpages/jammy/man1/yarn.1.html). This is why I used `npm` to install `yarn`.

Reply:
Updated.